Refactor and clarify "Blocking mode" page

breadtk opened the pull request:
This commit includes a gentle rewrite of the "Blocking Mode" docs page.
It reorders, expands, and clarifies each section with more detail about
the blocking modes that currently exist. It also includes an attempt to
standardizes/unifies each section's explanation to be more consistent
and clear.

Signed-off-by: Osman <osman@surkatty.org>

---
- **What does this PR aim to accomplish?:**

This commit is a gentle rewrite of the "[Blocking Mode](https://docs.pi-hole.net/ftldns/blockingmode/)" docs page.
It reorders, expands, and clarifies each section with more detail about
the blocking modes that currently exist. It also includes an attempt to
standardizes/unifies each section's explanation to be more consistent
and clear.

- **How does this PR accomplish the above?:**

I went through the page from top to bottom looking for ways to clarify each mode, reordered sentences, and clarified steps required to apply each mode. I also verified in code that there are no other known blocking modes defined (see [enums.h](https://github.com/pi-hole/FTL/blob/d10b6a6c0b1d807ec23ddc3b97c16d4dae67ee57/src/enums.h#L77-L83) and [gravity.sh](https://github.com/pi-hole/pi-hole/blob/f998eee4675c92819bde40f338fefe8feb7ba3b4/gravity.sh#L603-L623)).


- **What documentation changes (if any) are needed to support this PR?:**

None, this is the documentation! :)


---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

yubiuser commented on docs/ftldns/blockingmode.md:
```diff
@@ -21,52 +25,50 @@ doubleclick.net.        2       IN      A       0.0.0.0
 
 ## Pi-hole's full IP blocking
 
+In `IP` mode, blocked queries will be answered with the local IP addresses of your Pi-hole (see [BLOCK_IP4](configfile.md#block_ipv4) and [BLOCK_IP6](configfile.md#block_ipv6) for additional options). To set this mode explicitly, include `BLOCKINGMODE=IP` in the `/etc/pihole/pihole-FTL.conf` file and restart the `pihole-FTL` process.
```
```suggestion
In `IP` mode, blocked queries will be answered with the local IP addresses of your Pi-hole (see [BLOCK_IP4](configfile.md#block_ipv4) and [BLOCK_IP6](configfile.md#block_ipv6) for additional options). To set this mode explicitly, set `BLOCKINGMODE=IP` in `/etc/pihole/pihole-FTL.conf`.
```

yubiuser commented on docs/ftldns/blockingmode.md:
````diff
@@ -77,20 +79,20 @@ doubleclick.net.        2       IN      A       192.168.2.11
-```
-BLOCKINGMODE=NXDOMAIN
-```
+In `NXDOMAIN` mode, blocked queries will be answered with an empty response (i.e., there won't be an *answer* section) and status `NXDOMAIN`. A `NXDOMAIN` response should indicate that there is *no such domain* to the client making the query. To set this mode explicitly, include `BLOCKINGMODE=NXDOMAIN` in the `/etc/pihole/pihole-FTL.conf` file and restart the `pihole-FTL` process.
````
```suggestion
In `NXDOMAIN` mode, blocked queries will be answered with an empty response (i.e., there won't be an *answer* section) and status `NXDOMAIN`. A `NXDOMAIN` response should indicate that there is *no such domain* to the client making the query. To set this mode explicitly, set `BLOCKINGMODE=NXDOMAIN` in `/etc/pihole/pihole-FTL.conf`.
```

yubiuser commented on docs/ftldns/blockingmode.md:
````diff
@@ -99,17 +101,13 @@ Blocked queries will be answered with an empty response (no answer section) and
-```
-BLOCKINGMODE=NODATA
-```
+In `NODATA` mode, blocked queries will be answered with an empty response (no answer section) and status `NODATA`. A `NODATA` response indicates that the domain exists, but there is no record for the requested query type. To set this mode explicitly, include `BLOCKINGMODE=NODATA` in the `/etc/pihole/pihole-FTL.conf` file.
````
```suggestion
In `NODATA` mode, blocked queries will be answered with an empty response (no answer section) and status `NODATA`. A `NODATA` response indicates that the domain exists, but there is no record for the requested query type. To set this mode explicitly, set `BLOCKINGMODE=NODATA` in `/etc/pihole/pihole-FTL.conf`.
```

breadtk commented:
> In general, I'm not sure if I like the change from a code block style for the setting parameter to a in-line style...

I understand where you're coming from and was trying to think of a reasonable path forward as well. Where I ended up was that having a code block which includes a single line change seemed a bit much, especially when the documentation uses a code block in one section to signify contents of file, but a few lines later as console output. I figured it would tighten up the documentation a bit by removing the file content block and keep the query output instead. However, I am happy to change it back if you think this is a blocking change. Let me know either way.

Re: other suggestions, I really like your changes and I agree it makes everything else flow better. I appreciate the suggestions.

yubiuser commented:
It's not a blocking change. I just read the whole page again in I think we can keep it like it is now.
___

While you're on it - could you change the style for `advantages` and `disadvantages` from heading (`###`) to just bold (`**`). Reason is that all headings are added to the TOC which looks quite ugly at this page

![Bildschirmfoto zu 2022-09-07 22-13-26](https://user-images.githubusercontent.com/26622301/188972237-eed10052-a861-4989-9c9c-2b252c573d3d.png)
____

And please squash the majority of the commits.

PromoFaux commented:
![image](https://user-images.githubusercontent.com/1998970/188972922-4725592a-7a0a-439b-a3a0-7606b53ccfc5.png)

Is this section supposed to be there? It is not in the other headings

breadtk commented:
Thanks for the feedback you PromoFaux and yubiuser. I've gone ahead and incorporated the feedback and squashed my commits. Let me know if there is anything else required.

breadtk commented:
Looks like I had originally messed up the squash. That should be fixed now though. Let me know if you have questions/suggestions still. Cheers.

yubiuser reviewed (changes requested):
Only one last minor inconsistency: Twice you wrote "blocking mode" and three times only "blocking".

breadtk commented:
Updated! Anything else?

yubiuser commented:
Thanks for your PR, looking forward for your next one :-)

breadtk commented:
Thank you for the merge. You folks do a great job on this project and I can’t wait to see what you come up with next. :)
